### ISSUE TITLE
permissions and names with whitespace

### ISSUE THREAD

**Issue by midspace** [midspace/Space-Engineers-Admin-script-mod#68]
WolfNacht44
```
Just a side note, is there any possibility for Steam Names with Whitespace in their names, IE: Wolf Nacht 44, Currently trying to add a player to a group and function isnt detecting player name. however it recognises names without whitespace no problem.
```

**Comment by midspace** [comment]
The above may have to interpret the player hotlist before adding to the dictionary to quickly overcome this problem, but overall I think we need to address the whitespace issue for commands.

We need a better approach for splitting commands with quote delimited strings in the content before having to resort to hot lists for everything.

Players from what I can see in Steam could have double quotes ("), single quotes (') and spaces in their names.
Thankfully it doesn't look like names can be padded with spaces on either end.
Faction and ship names however, I'm not so sure about.

Added complexity, is that different countries have differing standards for what they use for quotes (partially determined by the keyboard).
This article indicates more: https://en.wikipedia.org/wiki/Quotation_mark

A player for instance could have: Harry "magic" Houdini
we could mistakenly trap ["magic"] as the name, when instead it should be [Harry "magic" Houdini].

This it when it comes down to detecting nested quotes.
ie.,
/kick 'Harry "magic" Houdini'
/kick [Ronald "Mc'Donald" Clown]

**Comment by Spcemarine** [comment]
Well, I think without a form of escaping, we can't provide a method that works regardless what the playername is. But I tend to point to the player lists rather than implement escaping because there are playernames that simply can't be written with a normal keyboard. E.g. some players tend to use Cyrillic characters, maybe because they are Russians or maybe because they think it looks fancy. Or use other unicode character so there will never be a way to type playernames...

Well, just some thoughts of a very lazy person like me.

However, if we gonna do it, we need to be able to escape signs. E.g. to kick a player named [ClanTag] Ronald "Mc'Donald Clown you'öö meed to type /kick "[ClanTag] Ronald \"Mc'Donald\" Clown". Too difficult for most users, I'd say, but still a way to go. Like I said, lists are much easier. ;)

**Comment by zrisher** [comment]
Agreed, pretty much every other field you can introduce a delimiter people can easily use in commands to contain delineate parameters.

Names, though, can have any funky combination of unicode characters they want.

**Comment by jpcsupplies** [comment]
pattern matching like the ore names?  To kick Billy "The Griefer"   use /kick billy
It scans the connected players list, finds only one "Billy....."  online and kicks him no matter what funky nickname he has.

Pretty useless for offline contexts - unless you maintain a list (hello econ mod!)  with a uid too.  (maybe add a seen command to admin script so you have a uid list *chuckles*)

But for one hit commands directed at online players, shouldn't pattern matching be adequate, eg kicking adding to groups etc.

**Comment by midspace** [comment]
The lists work. My understanding of the permissions wasn't very good when I wrote the above.
I have since revised my knowledge when I went through and created the wiki pages.
https://github.com/midspace/Space-Engineers-Admin-script-mod/wiki/Permissions

I do think we need to improve the quotes detection, but the effort to make a near perfect system isn't worthwhile for either us, or the admin who may attempt to type in a player's name, compared to simply using the hotlist.

Matintaing a list of previously connected players may be a good idea however.
